cds_ils/circulation/indexer.py

```diff
@@ -18,17 +19,25 @@ def index_extra_fields_for_loan(loan_dict):
     needed when fetching the loan details.
     """
     document_class = current_app_ils.document_record_cls
-    document_record = document_class.get_record_by_pid(
-        loan_dict.get("document_pid")
-    )
+    try:
+        document_record = document_class.get_record_by_pid(
+            loan_dict["document_pid"]
+        )
+    except PIDDeletedError:
+        # Document might have been deleted while reindexing asynchronously.
+        return
+
     document = document_record.replace_refs()
 
-    on_shelf = document.get("items", []).get("on_shelf", {})
+    on_shelf = document["items"]["on_shelf"]
 
+    # There may be no items on shelf, making impossible to suggest an item.
     if not on_shelf:
         loan_dict["item_suggestion"] = {}
         return
 
+    # The item_suggestion field is obtained by selecting the first library that 
```

Comment:
I meant more to describe the reason why we are doing it (adding the item_suggestion) in the first place